lib/api.js

```diff
@@ -57,15 +57,14 @@ api.declare({
   title: 'Run an instance',
   stability: API.stability.experimental,
   input: 'run-instance-request.json#',
-  scopes: [['ec2-manager:manage-resources:<workerType>']],
-  deferAuth: true,
+  scopes: 'ec2-manager:manage-resources:<workerType>',
   description: [
     'Request an instance of a worker type',
   ].join(' '),
 }, async function(req, res) {
   try {
     let workerType = req.params.workerType;
-    if (!req.satisfies({workerType: workerType})) { return undefined; }
+    await req.authorize({workerType: workerType});
```

Comment:
It's sort of scary to me that we're doing this as a promise, since forgetting an await on this promise would allow this code to silently not block the method from running, and would not generate an unhandled promise rejection error (I think).

Reply:
Yeah, it kinda scares me too. After a good bit of thinking on this,  I think it will actually create an `unhandledRejection` and we send alerts to sentry every time this happens so it will be caught quickly when it happens. For extra safety we could try adding https://github.com/mcollina/make-promises-safe as well?

Reply:
I'm also trying to think of some other mechanism by which we can ensure that it is awaited but not having much success yet. Unfortunately it needs to talk to auth so it has to be async.